src/components/Carousel.tsx

```diff
@@ -1,18 +1,186 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+interface Props {
+  images: string[];
+}
+
+const Carousel: React.FC<Props> = ({ images }) => {
+  const [translate, setTranslate] = useState(0);
+  const [itemWidth, setItemWidth] = useState(130);
+  const [frameSize, setFrameSize] = useState(3);
+  const [step, setStep] = useState(3);
+  const [animationDuration, setAnimationDuration] = useState(1000);
+  const [infinityMode, setInfinityMode] = useState(false);
+
+  const maxStep = 100 - ((100 / images.length) * frameSize);
+  const currentStep = (100 / images.length) * step;
+
+  function nextClick() {
+    setTranslate(prevTranslate => {
+      if (prevTranslate === maxStep) {
+        return 0;
+      }
+
+      return prevTranslate + currentStep > maxStep
+        ? maxStep
+        : prevTranslate + currentStep;
+    });
+  }
+
+  function prevClick() {
+    setTranslate(prevTranslate => {
+      if (prevTranslate === 0) {
+        return maxStep;
+      }
+
+      return prevTranslate - currentStep < 0
+        ? 0
+        : prevTranslate - currentStep;
+    });
+  }
+
+  function isFirstImage() {
+    return translate <= 0;
+  }
+
+  function isLastImage() {
+    return translate >= maxStep;
+  }
```

Comment:
```suggestion
  const isLastImage =   translate >= maxStep;
```
looks like redundant to make function in these cases.